src/index.js

```diff
@@ -1302,15 +1309,20 @@ const initializeFormElements = () => {
 
   // Malicious ERC20 Approval
   maliciousApprovalButton.onclick = async () => {
+    let erc20Contract;
+
+    if (networkName) {
+      erc20Contract = ERC20_SAMPLE_CONTRACTS[networkName];
+    } else {
+      erc20Contract = '0x4fabb145d64652a948d72533023f6e7a623c7c53';
+    }
     const result = await ethereum.request({
       method: 'eth_sendTransaction',
       params: [
         {
           from: accounts[0],
-          to: '0x4fabb145d64652a948d72533023f6e7a623c7c53',
-          gas: '0x30d40',
+          to: erc20Contract,
           data: '0x095ea7b3000000000000000000000000e50a2dbc466d01a34c3e8b7e8e45fce4f7da39e6000000000000000000000000000000000000000000000000ffffffffffffffff',
-          gasPrice: '0x76c3b0342',
```

Comment:
we remove suggested gas to avoid gas issues, so gas is calculated based on the network we are - we don't need to change this for each network